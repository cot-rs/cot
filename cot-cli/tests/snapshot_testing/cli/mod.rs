use tempfile::TempDir;

use super::*;
#[test]
fn no_args() {
    insta::with_settings!(
        { filters => GENERIC_FILTERS.to_owned() },
        { assert_cmd_snapshot!(cot_cli!("cli")) }
    );
}

#[test]
fn manpages() {
    let tempdir = TempDir::new().unwrap();
    insta::with_settings!(
        { filters => [GENERIC_FILTERS, TEMP_PATH_FILTERS].concat() },
        { assert_cmd_snapshot!(cot_cli!("cli", "manpages", "-o", tempdir.path())) }
    );
}

#[test]
fn completions_missing_shell() {
    insta::with_settings!(
        { filters => [GENERIC_FILTERS, TEMP_PATH_FILTERS].concat() },
        { assert_cmd_snapshot!(cot_cli!("cli", "completions")) }
    );
}

#[test]
fn completions_bash() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "bash"));
}

#[test]
fn completions_elvish() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "elvish"));
}

#[test]
fn completions_fish() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "fish"));
}

#[test]
fn completions_powershell() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "powershell"));
}

#[test]
fn completions_zsh() {
    assert_cmd_snapshot!(cot_cli!("cli", "completions", "zsh"));
}
