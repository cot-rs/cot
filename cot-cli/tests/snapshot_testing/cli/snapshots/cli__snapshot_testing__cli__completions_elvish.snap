---
source: cot-cli/tests/snapshot_testing/cli/mod.rs
info:
  program: cot
  args:
    - cli
    - completions
    - elvish
---
success: true
exit_code: 0
----- stdout -----

use builtin;
use str;

set edit:completion:arg-completer[cot] = {|@words|
    fn spaces {|n|
        builtin:repeat $n ' ' | str:join ''
    }
    fn cand {|text desc|
        edit:complex-candidate $text &display=$text' '(spaces (- 14 (wcswidth $text)))$desc
    }
    var command = 'cot'
    for word $words[1..-1] {
        if (str:has-prefix $word '-') {
            break
        }
        set command = $command';'$word
    }
    var completions = [
        &'cot'= {
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
            cand -V 'Print version'
            cand --version 'Print version'
            cand new 'Create a new Cot project'
            cand migration 'Manage migrations for a Cot project'
            cand cli 'Manage Cot CLI'
            cand help 'Print this message or the help of the given subcommand(s)'
        }
        &'cot;new'= {
            cand --name 'Set the resulting crate name (defaults to the directory name)'
            cand --cot-path 'Use `cot` from the specified path instead of a published crate'
            cand --use-git 'Use the latest `cot` version from git instead of a published crate'
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
        }
        &'cot;migration'= {
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
            cand list 'List all migrations for a Cot project'
            cand make 'Generate migrations for a Cot project'
            cand help 'Print this message or the help of the given subcommand(s)'
        }
        &'cot;migration;list'= {
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
        }
        &'cot;migration;make'= {
            cand --app-name 'Name of the app to use in the migration (default: crate name)'
            cand --output-dir 'Directory to write the migrations to (default: migrations/ directory in the crate''s src/ directory)'
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
        }
        &'cot;migration;help'= {
            cand list 'List all migrations for a Cot project'
            cand make 'Generate migrations for a Cot project'
            cand help 'Print this message or the help of the given subcommand(s)'
        }
        &'cot;migration;help;list'= {
        }
        &'cot;migration;help;make'= {
        }
        &'cot;migration;help;help'= {
        }
        &'cot;cli'= {
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
            cand manpages 'Generate manpages for the Cot CLI'
            cand completions 'Generate completions for the Cot CLI'
            cand help 'Print this message or the help of the given subcommand(s)'
        }
        &'cot;cli;manpages'= {
            cand -o 'Directory to write the manpages to [default: current directory]'
            cand --output-dir 'Directory to write the manpages to [default: current directory]'
            cand -c 'Create the directory if it doesn''t exist'
            cand --create 'Create the directory if it doesn''t exist'
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
        }
        &'cot;cli;completions'= {
            cand -v 'Increase logging verbosity'
            cand --verbose 'Increase logging verbosity'
            cand -q 'Decrease logging verbosity'
            cand --quiet 'Decrease logging verbosity'
            cand -h 'Print help'
            cand --help 'Print help'
        }
        &'cot;cli;help'= {
            cand manpages 'Generate manpages for the Cot CLI'
            cand completions 'Generate completions for the Cot CLI'
            cand help 'Print this message or the help of the given subcommand(s)'
        }
        &'cot;cli;help;manpages'= {
        }
        &'cot;cli;help;completions'= {
        }
        &'cot;cli;help;help'= {
        }
        &'cot;help'= {
            cand new 'Create a new Cot project'
            cand migration 'Manage migrations for a Cot project'
            cand cli 'Manage Cot CLI'
            cand help 'Print this message or the help of the given subcommand(s)'
        }
        &'cot;help;new'= {
        }
        &'cot;help;migration'= {
            cand list 'List all migrations for a Cot project'
            cand make 'Generate migrations for a Cot project'
        }
        &'cot;help;migration;list'= {
        }
        &'cot;help;migration;make'= {
        }
        &'cot;help;cli'= {
            cand manpages 'Generate manpages for the Cot CLI'
            cand completions 'Generate completions for the Cot CLI'
        }
        &'cot;help;cli;manpages'= {
        }
        &'cot;help;cli;completions'= {
        }
        &'cot;help;help'= {
        }
    ]
    $completions[$command]
}

----- stderr -----
